pkg/test/application/echo/echo.go

```diff
@@ -206,7 +226,10 @@ func (s *httpServer) start() error {
 	}
 
 	// Start serving HTTP traffic.
-	go func() { _ = s.server.Serve(listener) }()
+	go func() {
+		wg.Done()
+		_ = s.server.Serve(listener)
```

Comment:
FYI: The window is probably too small, but there is still a race here between Done and serve. Practically speaking it probably won't  have an impact

Reply:
Agreed, and that was my though as well.  Unfortunately, I'm not sure how to reliably account for that with the golang http APIs.